compiler/damlc/lib/DA/Cli/Damlc/DependencyDb.hs

```diff
@@ -246,24 +279,143 @@ validatePkgId pkgId = do
         fail $ "Invalid package ID dependency in daml.yaml: " <> pkgId
     pure $ LF.PackageId $ T.pack pkgId
 
+-- Package resolution
+---------------------
+
+data FullPkgName = FullPkgName
+    { pkgName :: LF.PackageName
+    , pkgVersion :: LF.PackageVersion
+    } deriving (Eq, Ord, Show)
+
+data LockFile = LockFile
+    { dependencies :: [DependencyInfo]
+    } deriving Generic
+instance Aeson.FromJSON LockFile
+instance Aeson.ToJSON LockFile
+data DependencyInfo = DependencyInfo
+    { name :: String
+    , version :: String
```

Comment:
Any reason not to store `LF.PackageName`, `LF.PackageVersion` and `LF.PackageId` here? Would save you a few conversions. Getting a JSON instance for those should be easy if we don’t already have one (just gnd it)

Reply:
True, too many conversions. I'll change it.